lib/puppet/provider/sshkey/parsed.rb

```diff
@@ -33,10 +33,10 @@ def title
   end
 
   def self.default_target
-    case Facter.value(:operatingsystem)
+    case Facter.value('os.name')
     when 'Darwin'
       # Versions 10.11 and up use /etc/ssh/ssh_known_hosts
-      version = Facter.value(:macosx_productversion_major)
+      version = Facter.value('os.macos.version.major')
```

Comment:
Based on https://github.com/puppetlabs/puppet-lint/blob/d0e55603755b0e5eb63b8601d117f4628f925ed1/lib/puppet-lint/plugins/legacy_facts/legacy_facts.rb#L67C4-L67C31 I think this might need to be?

```suggestion
      version = Facter.value('os.macosx.version.major')
```

Reply:
good catch!